internal/http/update.go

```diff
@@ -36,11 +39,13 @@ func (h updateHandler) Routes(r chi.Router) {
 }
 
 func (h updateHandler) getLatest(w http.ResponseWriter, r *http.Request) {
-	latest := h.service.GetLatestRelease(r.Context())
-	if latest != nil {
-		render.Status(r, http.StatusOK)
-		render.JSON(w, r, latest)
-		return
+	if h.config.CheckForUpdates {
```

Comment:
This should be inverted, and return NoContent or similar to prevent errors going forward (and burrito code).